### PR TITLE
video/vnc: add vnc_fb_register

### PR DESCRIPTION
**Pull request by jianglianfang**
## Summary
Optimize the initialization of vnc, change it to vnc_fb_register.

## Impact
Nuttx vnc

## Testing
run nuttx with vnc

### CODE REVIEW

File: boards/arm/samv7/samv71-xult/src/sam_bringup.c [hunk 1]
```diff
@@ -61,6 +61,10 @@
 #  include <nuttx/leds/userled.h>
 #endif
 
+#ifdef CONFIG_VNCSERVER
+#include <nuttx/video/vnc.h>
```
Comment (xiaoxiang781216) [inline boards/arm/samv7/samv71-xult/src/sam_bringup.c:65]:
```suggestion
#  include <nuttx/video/vnc.h>
```

File: boards/sim/sim/sim/src/sim_bringup.c [hunk 2]
```diff
@@ -61,6 +61,10 @@
 #include <nuttx/lcd/lcd_dev.h>
 #endif
 
+#ifdef CONFIG_VNCSERVER
+#include <nuttx/video/vnc.h>
```
Comment (xiaoxiang781216) [inline boards/sim/sim/sim/src/sim_bringup.c:65]:
```suggestion
#   include <nuttx/video/vnc.h>
```

**Comment by jianglianfang** [comment]
I successfully compiled sim:vncserver in the docker [ https://github.com/apache/nuttx/pkgs/container/nuttx%2Fapache-nuttx-ci-linux],
Proceed as follows：
1. sudo docker run -it -v $PWD:/home/nuttx ghcr.io/apache/nuttx/apache-nuttx-ci-linux:latest@sha256:324c3660ddcc78fd7efcee52b80bc8060f007da44a3f51b7551bb27ba16ab34a /bin/bash
**2. cd /home/nuttx/nuttx**
3. rm Make.defs
4. ln -s /home/nuttx/nuttx/tools/../boards/sim/sim/sim/scripts/Make.defs Make.defs
**5. make distclean
6. ./tools/configure.sh -l sim:vncserver
7. make clean;make**

**Comment by xiaoxiang781216** [comment]
I guess the error is due to linker doesn't move the unused fb_register from final image. It could be fixed by changing fb_register to macro or inline function.

**Comment by acassis** [comment]
@jianglianfang before we merge it, could you please confirm that this tutorial still working after your modifications: https://nuttx.apache.org/docs/latest/platforms/sim/sim/index.html#running-vnc-server ?

**Comment by acassis** [comment]
Could you please verify the compilation error?

File: graphics/nxmu/nxmu_start.c [hunk 3]
```diff
@@ -108,6 +112,22 @@ static int nx_server(int argc, char *argv[])
   dev->setpower(dev, ((3 * CONFIG_LCD_MAXPOWER + 3) / 4));
 
 #else /* CONFIG_NX_LCDDRIVER */
+# ifdef CONFIG_VNCSERVER
+  /* Initialize the VNC server */
+    int display;
```
Comment (pkarashchenko) [inline graphics/nxmu/nxmu_start.c:117]:
```suggestion
  int display;
```

File: boards/sim/sim/sim/src/sim_bringup.c [hunk 4]
```diff
@@ -61,6 +61,10 @@
...
+#endif
+
 #if defined(CONFIG_INPUT_BUTTONS_LOWER) && defined(CONFIG_SIM_BUTTONS)
 #include <nuttx/input/buttons.h>
```
Comment (pkarashchenko) [inline boards/sim/sim/sim/src/sim_bringup.c:69]:
```suggestion
#  include <nuttx/input/buttons.h>
```

File: boards/sim/sim/sim/src/sim_bringup.c (hunk 2 above)
Comment (pkarashchenko) [reply boards/sim/sim/sim/src/sim_bringup.c:65]: I would say 2 spaces

File: include/nuttx/video/fb.h [hunk 5]
```diff
@@ -1106,7 +1109,30 @@ int fb_register_device(int display, int plane,
  *
  ****************************************************************************/
 
-int fb_register(int display, int plane);
+static inline_function unused_code int fb_register(int display, int plane)
```
Comment (xiaoxiang781216) [inline include/nuttx/video/fb.h:1112]:
```suggestion
static inline_function int fb_register(int display, int plane)
```

File: graphics/nxmu/nxmu_start.c [hunk 6]
```diff
@@ -132,6 +152,7 @@ static int nx_server(int argc, char *argv[])
       return EXIT_FAILURE;
     }
 
+# endif /* CONFIG_VNCSERVER */
```
Comment (xiaoxiang781216) [inline graphics/nxmu/nxmu_start.c:155]:
```suggestion
#  endif /* CONFIG_VNCSERVER */
```

File: graphics/nxmu/nxmu_start.c [hunk 7]
```diff
@@ -108,6 +112,22 @@ static int nx_server(int argc, char *argv[])
...
+
+  display = atoi(argv[1]);
+
+  ret = vnc_fb_register(display);
+  if (ret < 0)
+    {
+       gerr("ERROR: vnc_fb_register() failed: %d\n", ret);
+    }
+
+# else /* CONFIG_VNCSERVER */
```
Comment (xiaoxiang781216) [inline graphics/nxmu/nxmu_start.c:129]:
```suggestion
#  else /* CONFIG_VNCSERVER */
```

File: graphics/nxmu/nxmu_start.c [hunk 8]
```diff
@@ -108,6 +112,22 @@ static int nx_server(int argc, char *argv[])
   dev->setpower(dev, ((3 * CONFIG_LCD_MAXPOWER + 3) / 4));
 
 #else /* CONFIG_NX_LCDDRIVER */
+# ifdef CONFIG_VNCSERVER
```
Comment (xiaoxiang781216) [inline graphics/nxmu/nxmu_start.c:115]:
```suggestion
#  ifdef CONFIG_VNCSERVER
```

File: graphics/nxmu/nxmu_start.c [hunk 9]
```diff
@@ -41,6 +41,10 @@
 
 #include "nxmu.h"
 
+#ifdef CONFIG_VNCSERVER
+#include <nuttx/video/vnc.h>
```
Comment (xiaoxiang781216) [inline graphics/nxmu/nxmu_start.c:45]:
```suggestion
#  include <nuttx/video/vnc.h>
```

File: boards/sim/sim/sim/src/sim_bringup.c (hunk 2 above)
Comment (xiaoxiang781216) [reply boards/sim/sim/sim/src/sim_bringup.c:65]: Yes.

File: graphics/nxmu/nxmu_start.c (hunk 9 above)
Comment (jianglianfang) [reply graphics/nxmu/nxmu_start.c]: done

File: graphics/nxmu/nxmu_start.c (hunk 3 above)
Comment (jianglianfang) [reply graphics/nxmu/nxmu_start.c]: done

File: include/nuttx/video/fb.h (hunk 5 above)
Comment (jianglianfang) [reply include/nuttx/video/fb.h]: without unused_code, there is a compilation error in macOS(sim-01) like Error: nxglib/nxglib_setpixel_1bpp.c:273:50: error: unused function 'fb_register' [-Werror,-Wunused-function]

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**